Beautify output and allow customization

milljm opened the pull request:
Darken the justification dots, and allow the user to apply
a format:

  export MOOSE_TERM_FORMAT='njcst'  # default config

  produces:
    simple_diffusion .... [CAVEAT] OK [0.000s]

Available formatting options:

'p' : pre result (status bucket result)
'n' : test name
'c' : caveats
'j' : justification
's' : status result
't' : test time

The skipped tests which have a reason (caveat) are now included as a caveat. The previous method looked odd.

New: 
```pre
test_name .................................... [METHOD!=DBG] SKIP
```
Old:
```pre
test_name .................................... skipped (METHOD!=DBG)
```


Closes #10453

Commit:
Beautify output and allow customization

Darken the justification dots, and allow the user to apply
a format:

  export MOOSE_TERM_FORMAT='njcst'  # default config

  produces:
    simple_diffusion .... [CAVEAT] OK [0.000s]

Available formatting options:

'p' : pre result (status bucket result)
'n' : test name
'c' : caveats
'j' : justification
's' : status result
't' : test time

Closes #10453

Commit:
Fix a unittest

Missed a location where I was not setting any caveats for
a skipped test.

Rebased for a missing unittest that needed to be altered.

Closes #10453

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/143683/) on 05eaf26 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/10454/site)

This comment will be updated on new commits.

milljm commented:
The recover tests look a little odd:
```pre
./run_tests --recover --re=actions/aux_scalar_variable.invalid_order_high
actions/aux_scalar_variable.invalid_order_high ............ [RECOVER] RunException RECOVER
actions/aux_scalar_variable.invalid_order_high_part1 ................ RunException RECOVER
```
They should mention they are skipped. This PR needs more work.

Commit:
Fixed several more SKIP caveats

These testers have skip logic, but did not apply a
caveat.

Closes #10453

milljm commented:
When reviewed, can I have some input on how the caveats for RECOVER parameters are displayed? I don't like the repeat text that is occurring:
```pre
misc/check_error.function_file_test11 ............. [RECOVER,RUNEXCEPTION RECOVER] SKIP
```

Normally this would be printed as:
```pre
misc/check_error.function_file_test11 ............. [RECOVER] skipped(RunException RECOVER)
```
Suggestions?

brianmoose commented on python/TestHarness/TestHarness.py:
```diff
@@ -393,40 +393,37 @@ def canPrint(self, tester):
 
-    # Format the caveats contained in tester so they are easy to read when printed
+    # Format results the way we want it
+    # TODO: refactore. It seems we only need it so the word FAILED
```
How involved would this refactor (not refactore) be? Something that would be suitable for this PR?

milljm replied:
This is something I will certainly be addressing in #10456

brianmoose commented on python/TestHarness/tests/test_Allocations.py:
```diff
@@ -7,16 +7,17 @@ def testSkippedAllocations(self):
-        output = self.runTests('-i', 'always_ok', '-p', '2', '-j', '1')
-        self.assertIn('skipped (insufficient slots)', output)
+        output = self.runTests('--no-color', '-i', 'always_ok', '-p', '2', '-j', '1')
+        #self.assertIn('skipped (insufficient slots)', output)
```
Can this just be removed?

brianmoose commented on python/TestHarness/tests/test_QueuePBS.py:
```diff
@@ -149,18 +149,21 @@ def _mockSinglePBSLaunch(self, harness, job, fake_result, mock_result):
-                # never runs jobB, but _occurres_ during jobA's iteration).
+                # A different method may have adjusted this tester to be
+                # skipped (jobA fails and thus skips and never runs jobB,
+                # but _occurres_ during jobA's iteration).
```
Is this supposed to be "_occurs_"?

brianmoose commented on python/TestHarness/tests/test_QueuePBS.py:
```diff
@@ -253,8 +257,9 @@ def testPBSBadQstat(self):
                 # Join the results, because the TestHarness receives them out-of-order
                 if job_id == 0:
                     join_results = ' '.join([result_list[0], result_list[1]])
+                    print join_results
```
`print` should use `()`

Commit:
Fix some typos to address comments

Addressing comments...

Remove a users MOOSE_TERM_FORMAT environment while
performing unittests.

Closes #10453

Commit:
Fixed some justification issues

Do not print duplicate status when using pre-result formatting.
This is primarily eye-candy, so Civet can use the new format and
support the displaying of 'RUNNING...' without all the duplicate
results.

Closes #10453

Commit:
Allow caveats to overflow if last

Simplify how items are discovered as printable.
Closes #10453

Commit:
Do not upper case the caveats by default

Allow the caveats to be printed unmolested

Closes #10453

Commit:
Handle unknown formatting codes

Allow case specification (needed for unittests to pass while
allowing pretty civet printing).

The default case for caveats is upper case.

Closes #10453

brianmoose commented on python/TestHarness/util.py:
```diff
@@ -146,86 +154,94 @@ def formatResult(tester_data, result, options, color=True):
-    if tester.getCaveats() and 'c' in terminal_format:
+    formatted_results = {}
+    for format_key in terminal_format:
+        formatted_results[format_key] = None
```
These 3 lines can be done with:
`formatted_results = dict.fromkeys(terminal_format)`

brianmoose commented on python/TestHarness/util.py:
```diff
@@ -146,86 +154,94 @@ def formatResult(tester_data, result, options, color=True):
+        if str(f_key).lower() == 't' and options.timing:
+            actual = float(tester_data.getTiming())
+            int_len = len(str(int(actual)))
+            precision = min(3, max(0,(4-int_len)))
```
Could you add a comment what this does? I know because you told me but I probably won't remember in 6 months.

brianmoose commented on python/TestHarness/util.py:
```diff
@@ -6,7 +6,7 @@
 
 TERM_COLS = int(os.getenv('MOOSE_TERM_COLS', '110'))
-TERM_FORMAT = os.getenv('MOOSE_TERM_FORMAT', 'njcst')
+TERM_FORMAT = os.getenv('MOOSE_TERM_FORMAT', 'njCst')
```
Not sure if we want this to be the default. Specifically, the "C" means that all caveats will be upper case. So messages like `Requires libMesh/libmesh#1409` will be output as `REQUIRES LIBMESH/LIBMESH#1409`
Maybe @permcody has an opinion.

milljm replied:
@brianmoose I agree!

The only reason I made it the default, is because unittests is expecting a lot of caveats to be upper-case.

I can change that; make it look pretty for us by default, and alter the environment when unittests are run so it passes unittests... but to me that sounds a little wrong (the default should be passing unittests).

Unless I am being overly protective of unittesting... I have no problems making this change if the rest of us are okay with it.

It would be completely transparent to everyone. `./run_tests` while in moose/python will still work no matter what you have set as your preferred MOOSE_TERM_FORMAT.

Commit:
Make the caveats print normally by default

Make the pretty printing default. While instead modifying
the environment when performing unittests instead.

Add comments to what/why we are doing with timing decoration.

Simplify the way we construct formatted_results
based on incoming environment variables.

Closes #10453

brianmoose reviewed (dismissed):
@permcody You might want to look at this before merging.
All the jobs on this PR have `MOOSE_TERM_FORMAT=tpnsc` so you can see what it might look like

andrsd commented:
Look at this [job](https://www.moosebuild.org/job/145389/). `Test moose test` does not have the time, but `Test modules` does. It looks like `t` was not picked up...

milljm commented:
@andrsd 
Its because civet is not asking for time when we run moose tests:

```bash
./run_tests -j 12 -l 12
```

andrsd commented:
@milljm I see, `t` is used only when running with `-t`. Makes sense...

Commit:
Added a comment

Added a comment, to mostly trigger a new build so Codacy goes away

Closes #10453
